src/Components/Search/state.ts

```diff
@@ -0,0 +1,48 @@
+import { Container } from "unstated"
+
+interface StateContainer {
+  hasEnteredPreviews: boolean
+  selectedPreviewIndex: number
+  previewHrefs: string[]
+}
+
+export class SearchBarState extends Container<StateContainer> {
+  state = {
+    hasEnteredPreviews: false,
+    selectedPreviewIndex: 0,
+    previewHrefs: [],
+  }
+
+  incrementSelectedPreviewIndex() {
+    this.setState({
+      selectedPreviewIndex: this.state.selectedPreviewIndex + 1,
```

Comment:
Do we need to guard against overflowing the index?

Reply:
good call, we should. Will add the total count into this state so we can accomplish that.

Reply:
Oh nevermind all the items for a preview pane are registered so can just do `items.length`!